### PR TITLE
hide empty roomsublists when filtering via search/tagpanel

### PR DESCRIPTION
**Pull request by t3chguy**
# Fixes https://github.com/vector-im/riot-web/issues/6801

### CODE REVIEW

File: src/components/structures/RoomSubList.js [hunk 1]
```diff
@@ -347,8 +347,14 @@ var RoomSubList = React.createClass({
         var label = this.props.collapsed ? null : this.props.label;
 
         let content;
-        if (this.state.sortedList.length === 0 && !this.props.searchFilter && this.props.extraTiles.length === 0) {
-            content = this.props.emptyContent;
+        if (this.state.sortedList.length === 0) {
```
Comment (lukebarnard1) [inline src/components/structures/RoomSubList.js]: Should this not include `&& this.props.extraTiles.length === 0` ?

File: src/components/structures/RoomSubList.js [hunk 2]
```diff
@@ -347,8 +347,14 @@ var RoomSubList = React.createClass({
...
+            // check if emptyContent is defined, if falsey then don't show an empty sublist even without searchFilter
```
Comment (lukebarnard1) [inline src/components/structures/RoomSubList.js]: To me, this comment makes things more confusing. I would at least write it in the order of conditions, i.e. "If no search filter is applied and ..." or "Only show placeholder if ..."

File: src/components/structures/RoomSubList.js (hunk 1 above)
Comment (t3chguy) [reply src/components/structures/RoomSubList.js]: Yup you're right, mine worked by an assumption that the current uses of `<RoomSubList />` are the only possible use cases but in future, we may have alternate ones.

File: src/components/structures/RoomSubList.js (hunk 2 above)
Comment (t3chguy) [reply src/components/structures/RoomSubList.js]: Done thanks

File: src/components/structures/RoomSubList.js [hunk 3]
```diff
@@ -100,9 +100,11 @@ const RoomSubList = React.createClass({
...
     applySearchFilter: function(list, filter) {
         if (filter === "") return list;
-        return list.filter((room) => {
-            return room.name && room.name.toLowerCase().indexOf(filter.toLowerCase()) >= 0;
-        });
+        const lcFilter = filter.toLowerCase();
+        // case insensitive if room name includes filter,
+        // or if starts with `#` and one of room's aliases starts with filter
+        return list.filter((room) => (room.name && room.name.toLowerCase().includes(lcFilter)) ||
+            (filter[0] === '#' && room.getAliases().some((alias) => alias.toLowerCase().startsWith(lcFilter))));
```
Comment (lukebarnard1) [inline src/components/structures/RoomSubList.js]: This doesn't look like a merge commit, it includes changes to include matches against room aliases. 

That's fine, but let's squash this PR (when merging) for simplicity.

Comment (t3chguy) [reply src/components/structures/RoomSubList.js]: https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/components/structures/RoomSubList.js#L111

Comment (t3chguy) [reply src/components/structures/RoomSubList.js]: https://github.com/matrix-org/matrix-react-sdk/pull/1957/files

**Comment by t3chguy** [comment]
restaged things for you @lukebarnard1

**Pull request merged**